schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/dao/common/ActivityDao.java

```diff
@@ -46,16 +91,11 @@ public Key save(@Nonnull Activity entity, @Nonnull Key activityTypeId) throws Mo
         if (entity.getMaxSubscriptions() < 0) throw new FieldValueException("Activity.maxSubscriptions");
 
         ActivityType activityType = activityTypeDao.get(activityTypeId);
-        if (entity.getName() == null) {
-            entity.setName(activityType.getName());
-        }
 
-        if (entity.getId() == null) {
-            // New Activity
-            entity.setId(Datastore.allocateId(activityType.getOrganizationRef().getKey(), getMeta()));
+        String name = StringUtils.trimToNull(entity.getName());
+        if (StringUtils.isBlank(name)) {
+            entity.setName(activityType.getName());
```

Comment:
This part is correct but looking at the code closely I realise that if name is "abc   " it is not correctly handled so will fix that